hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/TestRMWebServicesNodeLabels.java

```diff
@@ -119,527 +132,442 @@ public TestRMWebServicesNodeLabels() {
   }
 
   @Test
-  public void testNodeLabels() throws JSONException, Exception {
+  public void testNodeLabels() throws Exception {
     WebResource r = resource();
```

Comment:
This WebResource is used all over the place. Maybe it could be reset in setUp, and use it as a member variable, which would result in the same behaviour, but would eliminate all the resource passing.